environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java

```diff
@@ -238,6 +242,13 @@ private void fireContainerInitializedEvent() {
      */
     public Event<Object> event() {
         checkState();
+        if (event == null) {
+            synchronized (this) {
```

Comment:
I am not following here - why do we want this in `sync` block?
Also, why don't we do it just once in sync block in the `complete` method?